litmus/director/TCID-DIR-OP-DELETE-EXTERNAL-CSPC-WITH-NO-VOLUME/test.yml

```diff
@@ -33,16 +33,16 @@
 
         ## Get node name
         - name: Get node name
-          shell: kubectl get bd -n openebs --no-headers | grep Unclaimed | awk '{print $2}' | tail -n 1
+          shell: kubectl get bd -n openebs -ojsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.labels.kubernetes\.io/hostname}{"\t"}{.status}{"\t"}{.spec.filesystem}{"\t"}{"\n"}{end}' | grep "Unclaimed" | grep -v "fsType" | awk {'print $2'} | tail -n 1
           register: node_name
 
-        ## Add bd name in cspc yml
-        - name: Add bd name in cspc yml
+        ## Add node name in cspc yml
+        - name: Add node  name in cspc yml
           shell: sed -i "s/nodeName/{{node_name.stdout}}/g" /utils/cspc.yml 
 
         ## Create cstorPoolOperation
         - name: Get blockdevice 
-          shell: kubectl get bd -n openebs --no-headers | grep Unclaimed | awk '{print $1}' | tail -n 1
+          shell: kubectl get bd -n openebs -ojsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.labels.kubernetes\.io/hostname}{"\t"}{.status}{"\t"}{.spec.filesystem}{"\t"}{"\n"}{end}' | grep "Unclaimed" | grep -v "fsType" | awk {'print $1'} | tail -n 1
```

Comment:
How you are verifying this blockdevice you are getting here is exactly from the node obtained from above. Here you are using tail that means the last one which means it can be BD from any node

Reply:
I am using same command for getting node name and bd name and after that by using awk command I am fetching the nodename and bd name @nsathyaseelan 
```amit-bhatt@Amit-Bhatt:~/e2e/oep-e2e-aws$ kubectl get bd -n openebs -ojsonpath='{range .items[*]}{.metadata.name}{"\t"}{.metadata.labels.kubernetes\.io/hostname}{"\t"}{.status}{"\t"}{.spec.filesystem}{"\t"}{"\n"}{end}'
blockdevice-0390814181a4f6c310095e3e3b07671a	ip-10-0-1-212.us-west-2.compute.internal	map[claimState:Claimed state:Active]	map[]	
blockdevice-03c5697384abb41ba2d4349628d049ad	ip-10-0-1-100.us-west-2.compute.internal	map[claimState:Claimed state:Active]	map[]	
blockdevice-0e76b129e60de6f801e9593d9a643a39	ip-10-0-1-136.us-west-2.compute.internal	map[claimState:Claimed state:Active]	map[]	
blockdevice-10786eefd655737ad0b9187aaf2a94f1	ip-10-0-1-212.us-west-2.compute.internal	map[claimState:Claimed state:Active]	map[]	
blockdevice-27cbfecacd6dc7eaa8ba6ab8c9c330cf	ip-10-0-1-212.us-west-2.compute.internal	map[claimState:Unclaimed state:Active]	map[]	
blockdevice-31d427e432190f28c60a7b75b475d7b2	ip-10-0-1-100.us-west-2.compute.internal	map[claimState:Claimed state:Active]	map[]	
blockdevice-3cb4de3de9de234019c7eaee35e4374b	ip-10-0-1-100.us-west-2.compute.internal	map[claimState:Unclaimed state:Active]	map[]	
blockdevice-3d6a943f46eae3d6c434b15138c05144	ip-10-0-1-212.us-west-2.compute.internal	map[claimState:Unclaimed state:Active]	map[]	
blockdevice-6a948744c4581cda467cb8c48b96bb91	ip-10-0-1-212.us-west-2.compute.internal	map[claimState:Unclaimed state:Active]	map[]	
blockdevice-6b141d927475befed9bc59c10a745cfc	ip-10-0-1-136.us-west-2.compute.internal	map[claimState:Claimed state:Active]	map[]	
blockdevice-702f63ccd5fc2eda91f92733eda76518	ip-10-0-1-100.us-west-2.compute.internal	map[claimState:Unclaimed state:Active]	map[]	
```